js/pChTest.js

```diff
@@ -0,0 +1,43 @@
+import { fetchAllChallenges } from './apiService.js';
+import Challenge from './challengeClass.js';
+
+const challengesListElement = document.querySelector(".challenges__list");
+
+let challengesArray = [];
+const displayedChallenges = 3;
+
+const displayNumberOfChallenges = async () => {
+    const { success, data: challenges, error } = await fetchAllChallenges();
+
+    if (!success) {
+        console.error("Failed to fetch challenges:", error);
+        challengesListElement.innerHTML = `<li>Error loading challenges. Please try again later.</li>`;
+        return;
+    }
+
+    challengesArray = challenges;
+
+    let ascendingArray = [];
+
+    ascendingArray = challengesArray.sort((b, a) => a.rating - b.rating);
+
+    for (let i = 0; i < displayedChallenges; i++) {
+        const challengeData = ascendingArray[i];
+
+        const challenge = new Challenge(
+            challengeData.id,
+            challengeData.title,
+            challengeData.description,
+            challengeData.type,
+            challengeData.minParticipants,
+            challengeData.maxParticipants,
+            challengeData.rating,
+            challengeData.image,
+            challengeData.labels
+        );
```

Comment:
One more addition:
![Screenshot_20241116_141716_Kiwi Browser](https://github.com/user-attachments/assets/ed881d69-08fa-412d-bb51-997ec9305cc7)

A function from the global window, .sort((b, a) => a.rating - b.rating) to sort challenges by rating
I'm not sure if it's necessary to move it to a separate file at the moment. 
Function has been tested and works.

Reply:
Great job on implementing the functionality to fetch and display challenges! Here are some suggestions to further improve the code:
1- Rename ascendingArray for clarity: Consider renaming ascendingArray to sortedChallenges to better reflect its purpose. Also, replace let with const since the variable is not reassigned.
2-  Instead of directly reassigning challengesArray, consider creating a sorted copy of the array using the spread operator like --- const sortedChallenges = [...challengesArray].sort((a, b) => b.rating - a.rating);
3- I would recommend exploring the use of Array methods like .forEach(), .slice(), and .map() as an alternative to the for loop. These methods can make the code more concise and functional. If you have the time, it could be a great opportunity to research and try them out (b).

Reply:
```
const sortedChallenges = [...challengesArray].sort((a, b) => b.rating - a.rating);

const challengeData = sortedChallenges[i];

const challengesToDisplay = sortedChallenges.slice(0, 3);
    for (let i = 0; i < challengesToDisplay.length; i++) {
        const challengeData = challengesToDisplay[i];
```

- [ Done] 
- [ Done] 
- [ Done] 
- 💯